### ISSUE TITLE
sp_DatabaseRestore @RestoreDiff = 0 should set @BackupPathDiff to NULL

### ISSUE THREAD

**Issue by ShawnCrocker** [BrentOzarULTD/SQL-Server-First-Responder-Kit#1681]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
When @BackupPathDiff is specified and no file is found an error is thrown.
Even when @RestoreDiff = 0

It's perfectly reasonable to not find any DIFFs, and in any case we shouldn't enumerate @BackupPathDiff when @RestoreDiff = 0

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via http://sqlfiddle.com**
Point @BackupPathDiff at an empty folder

**What is the expected behavior?**
No error

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
SQL Server 2016

**Comment by BrentOzar** [comment]
Makes sense. If you want it to run automatically, option 3 is good because sometimes it'll run and just not happen to find diffs.
